Userland/Libraries/LibJS/Bytecode/ASTCodegen.cpp

```diff
@@ -26,6 +26,10 @@ Optional<Bytecode::Register> ScopeNode::generate_bytecode(Bytecode::Generator& g
     for (auto& child : children()) {
         last_value_reg = child.generate_bytecode(generator);
     }
+    if (!last_value_reg.has_value()) {
+        last_value_reg = generator.allocate_register();
+        generator.emit<Bytecode::Op::Load>(*last_value_reg, js_undefined());
+    }
```

Comment:
This seems wrong. Why would a ScopeNode set its last value to undefined? This is used for BlockStatement AFAICT, see https://tc39.es/ecma262/#sec-block-runtime-semantics-evaluation.

Reply:
And it's gone. :)